### DISCUSSION TITLE
How to interop with JS

### DISCUSSION THREAD

**Discussion by elringus** [unoplatform/Uno.Wasm.Bootstrap#439] [Q&A] [answered]
Maybe I got it wrong, but I hoped to use Uno.Wasm.Bootstrap as an alternative to Blazor, but without all the web app overhead, so I can just compile a C# library to WASM and use it from JS. But it looks like it doesn't have a built-in JS interop layer? I'm looking for something similar to:

- https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript
- https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet

Is the project not designed for this or I'm missing something?

**Comment by jeromelaban** [comment, marked as answer]
Invoking methods from Javascript can be done with [`Module.mono_bind_static_method`](https://github.com/unoplatform/Uno.Wasm.Bootstrap/blob/efa25b1f560794df398135d3781949a05ff93629/src/Uno.Wasm.Bootstrap/WasmScripts/uno-bootstrap.js#L212).

There's no public API provided by .NET outside of blazor that allows for invoking Javascript, but [this class can be added](https://github.com/unoplatform/Uno.Wasm.Bootstrap/blob/main/src/Uno.Wasm.Tests.Shared/Runtime.cs) to compensate for that, Uno.UI also provides more advanced interop (with Async support).
